### PR TITLE
fix(TBD-11717): CVE on mongo-java-driver-3.2.1.jar and mongo-java-driver-3.4.2.jar, need upgrade in reference components

### PR DESCRIPTION
**Pull request by firefryy**
**Please check if the PR fulfills these requirements**

- [ ] The commit(s) message(s) follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format) ?
- [ ] Unit tests for the Java changes have been added (for bug fixes / features) ?
- [ ] TUJ for the JavaJet changes have been added (for bug fixes / features) ?
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [ ] The new code does not introduce new technical issues

**What is the current behavior?** (You can also link to an open issue here)

**What is the new behavior?**

**BREAKING CHANGE**

If this PR contains a breaking change, please describe the impact and migration
path for existing applications.
If not please remove this section.

**Other information**:

### CODE REVIEW

File: main/plugins/org.talend.designer.components.bigdata/components/tMongoDBConnection/tMongoDBConnection_java.xml [hunk 1]
```diff
@@ -169,7 +169,6 @@
     <CODEGENERATION>
         <IMPORTS>
             <IMPORT MODULE="mongo-java-driver-3.12.0.jar" MVN="mvn:org.mongodb/mongo-java-driver/3.12.0"  NAME="mongo-java-driver-3.12.0"  REQUIRED_IF="(DB_VERSION != 'MONGODB_2_6_X') AND (DB_VERSION != 'MONGODB_2_5_X')"/>
-            <IMPORT MODULE="mongo-java-driver-3.4.2.jar" MVN="mvn:org.talend.libraries/mongo-java-driver-3.4.2/6.0.0"  NAME="mongo-java-driver-3.4.2" UrlPath="platform:/plugin/org.talend.libraries.mongo/lib/mongo-java-driver-3.4.2.jar" REQUIRED_IF="(DB_VERSION == 'MONGODB_2_6_X') OR (DB_VERSION == 'MONGODB_2_5_X')" />
```
Comment (hliashchuk) [inline main/plugins/org.talend.designer.components.bigdata/components/tMongoDBConnection/tMongoDBConnection_java.xml:172]: what driver will be use for 2.5 and 2.6 X?

Comment (hliashchuk) [reply main/plugins/org.talend.designer.components.bigdata/components/tMongoDBConnection/tMongoDBConnection_java.xml:172]: this relates to all changes in this PR?
I see that you haven't modified the required if condition in mongo-java-driver-3.4.2.jar, that's why I'm asking

Comment (firefryy) [reply main/plugins/org.talend.designer.components.bigdata/components/tMongoDBConnection/tMongoDBConnection_java.xml:172]: you're right, every version should use the new driver, corrected.

Comment (lbourgeois) [reply main/plugins/org.talend.designer.components.bigdata/components/tMongoDBConnection/tMongoDBConnection_java.xml:172]: I wonder why we had 2 different driver versions before if a single one works for all mongo versions

**Review by lbourgeois — APPROVED** [review]
LGTM if a single driver version can work for all mongo versions

**Pull request closed without merging**